pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BrokerRequestHandler.java

```diff
@@ -537,6 +537,8 @@ private BrokerResponse processOptimizedBrokerRequests(@Nonnull BrokerRequest ori
     long routingStartTime = System.nanoTime();
     Map<ServerInstance, SegmentIdSet> segmentServices = findCandidateServers(brokerRequest);
     phaseTimes.addToRoutingTime(System.nanoTime() - routingStartTime);
+
+    // TODO: need to differentiate the case of no server found because all segments are pruned.
```

Comment:
After the discussion with @Jackie-Jiang , we can keep the current behavior for the case of no server found vs no server found due to pruning. I initially thought that `no server found` means the failure case where no server is alive for a segment (So, it makes sense to throw an exception). However, this covers more generic cases (i.e. empty table). In case of all servers down for a segment, all of broker builder implementations would throw an exception so it won't go through this code path.